packages/ra-ui-materialui/src/input/SelectArrayInput.tsx

```diff
@@ -262,8 +274,8 @@ export const SelectArrayInput = (props: SelectArrayInputProps) => {
                             {selected
                                 .map(item =>
                                     (allChoices || []).find(
-                                        choice =>
-                                            getChoiceValue(choice) === item
+                                        // eslint-disable-next-line eqeqeq
+                                        choice => getChoiceValue(choice) == item
```

Comment:
same

this probably deserves some tests with nullish values as well

Reply:
I've been told to wait for the issue 😏